storage-provider/server/src/indexer/mod.rs

```diff
@@ -104,7 +105,15 @@ where
 }
 
 /// Prepares indexes of a raw piece.
-async fn piece_indexes<P>(location: P) -> Result<Vec<IndexRecord>, ServerError>
+///
+/// # Returns
+///
+/// Returns a Result containing:
+/// - Ok((Vec<Cid>, Vec<IndexRecord>)): A tuple of:
+///   - Vector of Content IDs (roots from v1 header)
+///   - Vector of index records containing CIDs with their offsets and sizes
+/// - Err(ServerError): If any I/O or parsing operations fail
```

Comment:
The error is somewhat self-explanatory, so there's no need to explain it.
```suggestion
/// Returns the root CIDs and the index records mapping CIDs to their respective offsets and sizes. 
```